aries_cloudagent/protocols/connections/v1_0/routes.py

```diff
@@ -145,14 +147,24 @@ class ConnectionsListQueryStringSchema(OpenAPISchema):
         ),
         example=ConnRecord.Role.REQUESTER.rfc160,
     )
+    tx_my_role = fields.Str(
+        description=" A list of my assigned connection role (AUTHOR/ENDORSER)",
+        required=False,
+        example="AUTHOR",
+    )
+    tx_their_role = fields.Str(
+        description="A list of their assigned connection role (AUTHOR/ENDORSER)",
+        required=False,
+        example="ENDORSER",
+    )
 
 
 class CreateInvitationQueryStringSchema(OpenAPISchema):
     """Parameters and validators for create invitation request query string."""
 
     alias = fields.Str(
         description="Alias",
-        required=False,
+        required=True,
         example="Barry",
```

Comment:
Yes, Not any specific reason for that. We will change the required of alias as False

Reply:
> @HarshMultani @swcurran @ankita-p17 ... I’ve been doing some thinking about the design of the Endorser api. I think that the current approach (creating a signed transaction and dealing with TAA inside the api) is not the best. The existing api’s that write to the ledger (create schema, create cred def, etc.) already have all this logic, so adding it to the Endorser API is duplication.
> 
> I think to keep separation of concerns and to keep the code DRY:
> • Update the ledger-updating api's (create schema, create cred def etc.) to either write directly to the ledger, or return a prepared (signed & TAA-accepted) transaction
> • The controller can then pass this transaction as an attachment to the /transactions/request-endorsement endpoint
> • (this endpoint then doesn't have to worry about anything other than packaging the attachment and sending to the Endorser for endorsement)
> • Once the endorsement is received, the controller can call another endpoint (e.g. /transactions/write-to-ledger) to write the endorsed transaction to the ledger
> • (This last step will involve some indy-specific work, adding the endorser signature to the indy transaction)
> 
> Since we already have code for preparing the indy transactions, I don't think we should add this to the endorser api's at all

Hi @ianco I agree with you, also, **/transaction/send-request** API is doing the same thing you explained. It gets the indy transaction in body, prepares endorsement transaction with input as an attachment, and sends to the Endorser for endorsement.
- Existing Schema, CredDef create and write Revocation entry APIs can be updated to return indy transaction prepared which can be consumed in above-mentioned API
- A new API /write-to-ledger is to be added for Author. This will take transaction_id as input and submit the transaction to the ledger with Endorser DID, Verkey
- For the controller to get the API events we have to write webhooks for endorsement APIs. On intended webhook call, transaction author's controller can call /write-to-ledger
- Also, the schema/ credDef option in /transaction/endorse were to demo and doesn't make the API specific to the transaction type. You can still send any type of message as an attachment to endorse.

I hope this is on the same lines you explained. Please correct if anything.

Reply:
Hi @HarshMultani I don't see a `/transaction/send-request` endpoint, did you mean `/transactions/create-request`?

The `/create-request` endpoint contains a lot of logic dealing with TAA, which I think is duplicated from other endpoints that can create transactions, like schema create etc.

Reply:
Yes @ianco,  I meant the endpoint - /transactions/create-request